### PR TITLE
bpo-45847: Port xxlimited and xxlimited_35 to PY_STDLIB_MOD (GH-29707)

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue45847

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1474,13 +1474,11 @@ def detect_modules(self):
             self.missing.append('_tkinter')
         self.detect_uuid()
 
-##         # Uncomment these lines if you want to play with xxmodule.c
-##         self.add(Extension('xx', ['xxmodule.c']))
+        # Uncomment the next line if you want to play with xxmodule.c
+#        self.add(Extension('xx', ['xxmodule.c']))
```
Comment (erlend-aasland) [inline setup.py:1478]: Should we leave this commented out?

File: configure.ac [hunk 2]
```diff
@@ -6215,6 +6214,12 @@ PY_STDLIB_MOD([_testmultiphase], [test "$TEST_MODULES" = yes])
 PY_STDLIB_MOD([_xxtestfuzz], [test "$TEST_MODULES" = yes])
 PY_STDLIB_MOD([_ctypes_test], [test "$TEST_MODULES" = yes], [], [], [-lm])
 
+dnl Template modules.
+PY_STDLIB_MOD_SIMPLE([_xxsubinterpreters])
```
Comment (tiran) [inline configure.ac]: ``_xxsubinterpreters`` is not a template. It's an experimental module with bindings to subinterpreter API.

File: Modules/Setup.stdlib.in [hunk 3]
```diff
@@ -131,6 +130,14 @@
 @MODULE__TESTBUFFER_TRUE@_testbuffer _testbuffer.c
 @MODULE__TESTINTERNALCAPI_TRUE@_testinternalcapi _testinternalcapi.c
 
+
+############################################################################
+# Template modules
+
+@MODULE__XXSUBINTERPRETERS_TRUE@_xxsubinterpreters _xxsubinterpretersmodule.c
+@MODULE_XXLIMITED_TRUE@xxlimited xxlimited.c
+@MODULE_XXLIMITED_35_TRUE@xxlimited_35 xxlimited_35.c
```
Comment (tiran) [inline Modules/Setup.stdlib.in]: I think these modules must be build as shared modules, too. They use the limited API.

File: configure.ac (hunk 2 above)
Comment (erlend-aasland) [reply configure.ac]: Right, assumption hits me again.

**Review by tiran — APPROVED** [review]

**Pull request merged**